File: 0001_two_sum/tow_sum.py
def two_sum(nums, target):
    num_map = {}  # 用来存储数值和对应的下标
    for i, num in enumerate(nums):
        complement = target - num  # 计算差值
        if complement in num_map:
            return [num_map[complement], i]  # 找到配对
        num_map[num] = i  # 没找到就记录当前数值和下标
    return []

# 示例：
nums = [2, 7, 11, 15]
target = 9
result = two_sum(nums, target)
print(result)  # 输出 [0, 1]

File: 0001_two_sum/tow_sum.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int) // 数值 -> 下标

	for i, num := range nums {
		complement := target - num
		if index, ok := numMap[complement]; ok {
			return []int{index, i} // 找到就返回两个下标
		}
		numMap[num] = i // 没找到就记录当前数字和下标
	}
	return nil // 如果没有找到（题目说一定有答案）
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	result := twoSum(nums, target)
	fmt.Println(result) // 输出：[0 1]
}

